### PR TITLE
Add fully functional trader side application

### PR DESCRIPTION
**Pull request by jason-nn**
Added all features for the trader side application of the admin app
***
Account page that shows the user's current balance
Screen Shot 2022-03-02 at 8 43 48 PM
***
Allow users to add funds to their account
Screen Shot 2022-03-02 at 8 43 52 PM
***
Transactions page that shows all transactions
Screen Shot 2022-03-02 at 8 43 57 PM
***
Stocks index page where users can view all stocks
Screen Shot 2022-03-02 at 8 43 30 PM
***
Stocks show page where users can buy and sell stocks
Screen Shot 2022-03-02 at 8 43 38 PM
***
Portfolio page where users can view all the stocks in their portfolio
Screen Shot 2022-03-02 at 8 43 44 PM

### CODE REVIEW

File: db/migrate/20220222150049_create_transactions.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ def change
     create_table :transactions do |t|
       t.string :action
       t.float :amount
-      t.string :stock
+      t.bigint :stock_id
```
Comment (analyn-cajocson) [inline db/migrate/20220222150049_create_transactions.rb]: Why do you need to modify this migration file? If it is already in the `main` branch better to create a new migration file and apply the needed update to your database structure.

File: db/seeds.rb [hunk 2]
```diff
@@ -5,3 +5,29 @@
...
+    email: 'jason.ho@obf.ateneo.edu',
+    password: 'password',
+    admin: false,
+  )
+
+client =
+  IEX::Api::Client.new(
+    publishable_token: 'pk_714d506665774bd4911a522edde00754',
+    secret_token: 'sk_5a1c52d71b9a4d1d880b1f44646e44af',
+    endpoint: 'https://cloud.iexapis.com/v1',
```
Comment (analyn-cajocson) [inline db/seeds.rb]: Your credential is confidential. Refrain from putting it in a plain text only.

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]

File: db/migrate/20220222150049_create_transactions.rb (hunk 1 above)
Comment (jason-nn) [reply db/migrate/20220222150049_create_transactions.rb]: Noted, thank you for the feedback. This has been resolved in this commit: https://github.com/jason-nn/batch14-stocks/pull/8/commits/87f6ab797e49c29650885f833e426f8adf7c1524

File: db/seeds.rb (hunk 2 above)
Comment (jason-nn) [reply db/seeds.rb]: What would you recommend as the best way to store keys in this project?

Comment (analyn-cajocson) [reply db/seeds.rb]: you can use `env` variables or the `credentials.yml.enc`

File: Gemfile [hunk 3]
```diff
@@ -62,4 +62,6 @@ end
 
 gem 'rails-controller-testing'
 
-gem 'devise'
\ No newline at end of file
+gem 'devise'
+
+gem 'iex-ruby-client'
```
Comment (analyn-cajocson) [inline Gemfile]: add new line at the end of file

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 3 above)
Comment (jason-nn) [reply Gemfile]: what's the purpose of adding a new line at the end of file?

File: db/seeds.rb (hunk 2 above)
Comment (jason-nn) [reply db/seeds.rb]: noted, will work on this now

File: Gemfile (hunk 3 above)
Comment (analyn-cajocson) [reply Gemfile]: to avoid this warning from github
image

**Review by analyn-cajocson — APPROVED** [review]

**Review by juanmiguelemilio — APPROVED** [review]

**Pull request merged**